### PR TITLE
Increase SFDCAccessController coverage and assertions

### PR DESCRIPTION
**Pull request by FishOfPrey**
Starting code coverage of SFDCAccessController by testAccessController was 54.4%. Many areas around testing negative cases for Object and Field level security were untested.

I've avoided adding a dedicated testing profile with no access and have instead used the standard ReadOnly profile.  This should reduce the impact on other orgs if the source is installed directly rather than as a package.

The test coverage is now at 93.2%. More importantly, there are assertions that each of the test scenarios is producing the expected outcome.

### CODE REVIEW

File: src/classes/testAccessController.cls [hunk 1]
```diff
@@ -38,6 +38,44 @@
 @isTest
 private class testAccessController {
 
+	// Baseline Contact fields that will be accessiable when using the ReadOnly profile and createFieldPermissions()
+	private static List<String> fieldsToCheck = new List<String>{'LastName','FirstName','Salutation','OtherStreet','OtherCity','OtherState','OtherPostalCode','OtherCountry','OtherLatitude','OtherLongitude','MailingStreet','MailingCity','MailingState','MailingPostalCode','MailingCountry','MailingLatitude','MailingLongitude','Phone','Fax','MobilePhone','HomePhone','OtherPhone','AssistantPhone','ReportsToId','Title','Department','AssistantName','LeadSource','Birthdate','Description','OwnerId','EmailBouncedReason','EmailBouncedDate','Jigsaw','CleanStatus', 'Email'};
```
Comment (capeterson) [inline src/classes/testAccessController.cls]: Jigsaw fields might go away with the data.com end of life. Please choose another standard field with a more certain future.

File: src/classes/testAccessController.cls [hunk 2]
```diff
@@ -97,14 +135,21 @@ private class testAccessController {
...
         
         try {
         	ESAPI.accessController().setSharingMode(null);
         	System.assert(false, 'Should never get here');
-        } catch (Exception e) {
+        } catch (SFDCAccessController.AccessControlDmlException e) {
+        	// should fail - so all good
+        }
+
+		try {
```
Comment (capeterson) [inline src/classes/testAccessController.cls]: weird indent

File: src/classes/testAccessController.cls [hunk 3]
```diff
@@ -355,4 +1148,81 @@ private class testAccessController {
...
+		}
+
+		return u;
+	}
+
+	private static List<FieldPermissions> createFieldPermissions(Id permissionSetId, boolean read, boolean edit) {
+
+		string sObjectType = 'Contact';
+
+		List<string> readWriteFields = new List<String>{'Account','Phone','Fax','MobilePhone','HomePhone','OtherPhone','AssistantPhone','Title','Department','AssistantName','LeadSource','Birthdate','Description','Jigsaw','CleanStatus'};
```
Comment (capeterson) [inline src/classes/testAccessController.cls]: Another place to swap our jigsaw field

**Review by capeterson — CHANGES_REQUESTED** [review]

File: src/classes/testAccessController.cls (hunk 1 above)
Comment (FishOfPrey) [reply src/classes/testAccessController.cls]: I've removed the Jigsaw and CleanStatus data.com fields. They were mostly present as the ReadOnly profile granted access and I was trying to stress the methods under testing with the largest number of fields that standard profile allows.

File: src/classes/testAccessController.cls (hunk 2 above)
Comment (FishOfPrey) [reply src/classes/testAccessController.cls]: I've found that VSCode has defaulted to tabs with a size of 4. That's created an odd mixture of the existing 4-space indents and the newer 4-tab indents that doesn't look right on Github.
I did look at converting the entire class to consistent 4-space indenting, but it also picked up a large number of changes on the existing lines. I suspect this was part of the formatting problem I encountered when converting to the SFDX format.

For now I've just corrected this particular instance and will save the consistent document indentation for a subsequent pull request. Or would you prefer it all in one go?

File: src/classes/testAccessController.cls (hunk 3 above)
Comment (FishOfPrey) [reply src/classes/testAccessController.cls]: As above, removed the Jigsaw and CleanStatus fields from the tests.

File: src/classes/testAccessController.cls (hunk 2 above)
Comment (capeterson) [reply src/classes/testAccessController.cls]: Save the whitespace diffs for a different PR please. Formatting changes fit better as their own verifiably formatting-only PR.

Comment (FishOfPrey) [reply src/classes/testAccessController.cls]: OK. I think the PR has been corrected now and doesn't include the class wide formatting changes.

**Comment by capeterson** [comment]
Remote branch is too far ahead of what this PR was intended. Closing in favor of #16's cherry picked subset.

**Pull request closed without merging**